app.py

```diff
@@ -584,3 +591,39 @@ def resize_divisible(self, image, max_size=1024, divisible=16):
 
 
 app = ImageGenerationService()
+
+
+@app.app.post("/api/v1/txt2img")
+@limiter.limit(API_RATE_LIMIT) # Update the rate limit
+async def txt2img_api2(request: Request, data: TextToImage):
+    return await app.txt2img_api(request, data)
+
+@app.app.post("/get_credentials")
+@limiter.limit(API_RATE_LIMIT) # Update the rate limit
+async def get_credentials(request: Request, data: TextToImage):
+    return await app.get_credentials(request, data)
+
+@app.app.post("/generate")
+@limiter.limit(API_RATE_LIMIT) # Update the rate limit
+async def generate(request: Request, data: TextToImage):
+    return await app.generate(request, data)
+
+@app.app.get("/get_validators")
+@limiter.limit(API_RATE_LIMIT) # Update the rate limit
+async def get_validators(request: Request, data: TextToImage):
+    return await app.get_validators(request, data)
+
+@app.app.post("/api/v1/img2img")
+@limiter.limit(API_RATE_LIMIT) # Update the rate limit
+async def img2img_api(request: Request, data: TextToImage):
+    return await app.img2img_api(request, data)
+
+@app.app.post("/api/v1/instantid")
+@limiter.limit(API_RATE_LIMIT) # Update the rate limit
+async def instantid_api(request: Request, data: TextToImage):
+    return await app.instantid_api(request, data)
+
+@app.app.post("/api/v1/controlnet")
+@limiter.limit(API_RATE_LIMIT) # Update the rate limit
+async def controlnet_api(request: Request, data: TextToImage):
```

Comment:
Data model is incorrect. For example, with /img2img, /controlnet it should be ImgToImg.